wrapanapi/systems/ec2.py

```diff
@@ -46,7 +78,7 @@ def unset_tag(self, key, value):
                                                Tags=[{"Key": key, "Value": value}])
 
 
-class EC2Instance(Instance, _TagMixin):
+class EC2Instance(_SharedMethodsMixin, Instance, _TagMixin):
```

Comment:
Q: Don't `Mixin` class usually come after the others? I.e.

```suggestion
class EC2Instance(Instance, _TagMixin, _SharedMethodsMixin):
```

Reply:
Well, I was wondering as well...But I thought it makes some sense with overriding abstract methods? In openstack and rhevm mixin is before entity...
https://github.com/ManageIQ/wrapanapi/blob/master/wrapanapi/systems/openstack.py
https://github.com/ManageIQ/wrapanapi/blob/master/wrapanapi/systems/rhevm.py#L160

Reply:
@mmojzis It appears you're correct, after a bit more research I found https://www.ianlewis.org/en/mixins-and-python

Thanks!

Reply:
Based on that article, however, I would expect the order to be 

```suggestion,
class EC2Instance(_TagMixin, _SharedMethodsMixin, Instance):
```

Since `Instance` is the `BaseClass`

Reply:
You are right. I will add docstring to SharedMethodsMixin and will change the order of Classes.

Reply:
@john-dupuy Fixed :+1: